pkg/install/error_test.go

```diff
@@ -0,0 +1,213 @@
+package install
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"errors"
+	"net/http"
+	"reflect"
+	"testing"
+
+	"github.com/Azure/go-autorest/autorest"
+	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/golang/mock/gomock"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	mock_resources "github.com/Azure/ARO-RP/pkg/util/mocks/azureclient/mgmt/resources"
+)
+
+func TestIsResourceQuotaExceededError(t *testing.T) {
+	for _, tt := range []struct {
+		name        string
+		inputErr    error
+		wantBool    bool
+		wantMessage string
+	}{
+		{
+			name:     "Is not a DetailedError",
+			inputErr: errors.New("☹"),
+		},
+		{
+			name:     "Does not contain original error",
+			inputErr: autorest.NewErrorWithError(nil, "", "", nil, ""),
+		},
+		{
+			name:     "Original is not a ServiceError",
+			inputErr: autorest.NewErrorWithError(errors.New("☹"), "", "", nil, ""),
+		},
+		{
+			name: "No code in ServiceError",
+			inputErr: autorest.NewErrorWithError(&azure.ServiceError{
+				Details: []map[string]interface{}{{
+					"code":    nil,
+					"message": nil,
+				}}}, "", "", nil, ""),
+		},
+		{
+			name: "Different code in ServiceError",
+			inputErr: autorest.NewErrorWithError(&azure.ServiceError{
+				Details: []map[string]interface{}{{
+					"code":    "AnotherCode",
+					"message": "Something happened",
+				}}}, "", "", nil, ""),
+		},
+		{
+			name: "Quota exceeded but no message in ServiceError",
+			inputErr: autorest.NewErrorWithError(&azure.ServiceError{
+				Details: []map[string]interface{}{{
+					"code":    "QuotaExceeded",
+					"message": nil,
+				}}}, "", "", nil, ""),
+		},
+		{
+			name: "Quota exceeded",
+			inputErr: autorest.NewErrorWithError(&azure.ServiceError{
+				Details: []map[string]interface{}{{
+					"code":    "QuotaExceeded",
+					"message": "Quota exceeded",
+				}}}, "", "", nil, ""),
+			wantBool:    true,
+			wantMessage: "Quota exceeded",
+		},
```

Comment:
I feel like these test cases add more burden than they add value.  I think I'd check two cases:
1- a "real life" quota exceeded message causes the expected return values
2- some other error does not.

Ditto for TestIsDeploymentActiveError.